Skills: add skill to help user select package + solver (#943)
LLNL/sundials: .agents/skills/decision-tree/references/method-and-linear-solvers.md

When the user is comparing `CV_BDF` against implicit `ARKStep`, use this rule of thumb:

- prefer `CV_BDF` when the variable-order multistep workflow is the main advantage and there is no sign that higher-order BDF stability or damping will be problematic
- prefer implicit `ARKStep` when one-step A-stable or L-stable behavior matters, when strongly diffusive fast modes should be damped aggressively, or when a parabolic semi-discretization is a poor fit for higher-order BDF behavior
- if `CV_BDF` is otherwise attractive but stability theory points toward the A-stable range, mention capping the BDF order at 2 as a concrete alternative

### IDA and IDAS

`IDA` and `IDAS` use variable-order BDF methods. The main decision is not the integration formula but the solve stack around it:

- how to obtain consistent initial conditions
- whether the Jacobian should be dense, banded, sparse, or matrix-free
- whether direct factorization is affordable
- whether a preconditioned Krylov method is needed

### ARKODE

Choose the stepper from the model structure:

- `ERKStep` for explicit RK

> - `ARKStep` for implicit DIRK or IMEX additive RK

Reviewer: ```suggestion
- `ARKStep` for implicit DIRK or IMEX additive RK, or for explicit RK when the application involves a non-identity mass matrix
```

> - `MRIStep` for multirate

Reviewer: ```suggestion
- `MRIStep` for multirate applications
```

Use a direct solver when factorization is affordable.

- dense direct: small dense Jacobians
- band direct: narrow-band Jacobians with known bandwidth
- sparse direct such as KLU: sparse Jacobians where fill-in is still manageable

Direct methods are usually the cleanest recommendation for small to moderate problems because they remove preconditioner design from the initial setup.

### Krylov Solvers

Use Krylov methods for large stiff problems, large sparse problems, or matrix-free workflows.

Recommended first choice:

- GMRES / SPGMR as the best generic default

Alternatives:

- FGMRES / SPFGMR when the preconditioner changes between iterations
- BiCGStab / SPBCG when storage matters more than GMRES flexibility
- TFQMR / SPTFQMR when GMRES storage or restart behavior is a concern
- PCG when the linear system is symmetric positive definite

If you recommend a Krylov solver for a hard stiff problem, also discuss the preconditioner. A Krylov recommendation without a preconditioner plan is often incomplete.

## When To Push For Jacobians Or Preconditioners

Push for a user Jacobian, Jacobian-vector routine, or preconditioner when:

- finite differences are too expensive

> - the Jacobian is sparse or banded and the structure is known

Reviewer: ```suggestion
- the Jacobian is sparse, since that cannot be approximated internally using finite differences
- the Jacobian is banded and the structure is known
```
